packages/console/src/components/ActionMenu/index.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { MouseEventHandler, ReactNode, useRef, useState } from 'react';
+import ReactModal from 'react-modal';
+
+import { ButtonProps } from '../Button';
+import ActionMenuButton from './ActionMenuButton';
+import * as styles from './index.module.scss';
+import usePosition from './use-position';
+
+export { default as ActionMenuItem } from './ActionMenuItem';
+
+type Props = {
+  children: ReactNode;
+  buttonProps: ButtonProps;
+  title?: ReactNode;
+};
+
+const ActionMenu = ({ children, buttonProps, title }: Props) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const anchorReference = useRef<HTMLDivElement>(null);
+  const overlayReference = useRef<HTMLDivElement>(null);
+
+  const {
+    position: { left, top },
+    mutate,
+  } = usePosition(anchorReference, overlayReference);
+
+  const handleClick: MouseEventHandler<HTMLButtonElement> = () => {
+    setIsOpen(true);
+    // Calc position after modal opened.
+    setTimeout(() => {
+      mutate();
+    });
+  };
+
+  return (
+    <div className={styles.actionMenu}>
+      <ActionMenuButton {...buttonProps} ref={anchorReference} onClick={handleClick} />
+      <ReactModal
```

Comment:
Personally don't like the transparent overlay modal solution, as the overlay blocks everything underneath. Thus make it impossible to interact with when the dropdown drops.
Would be better just use react portal to implement the dropdown menu.

Reply:
I think, the dropdown menu is designed to BLOCK everything, and is a common practice. The thing that does not block everything is called tooltip or toast.

Reply:
Discussed and I'm OK with it